en/05_Contributing/05_Coding_Conventions/01_PHP_Coding_Conventions.md

```diff
@@ -136,6 +136,133 @@ class MyClass extends Class
 }
 ```
 
+## Code style
+
+- Prioritise the human readability of code. Avoid "clever" code that aims to reduce the number of lines at the expense of readbability. This is particularly important when writing Javascript.
+- Avoid writing any new "magic" code as it is very hard to follow.
+- Within reason, favour larger files that have everything contained in them rather than multiple small files that require developers to jump around multiple files to follow a single code path.
```

Comment:
Considering we advise against this in PHP land (1 class per file) I think we should stick with a preference is for a single file to be used for a single class / object to make life easier to find. Sure the trade off is multiple files but most users would be used to that (i.e jumping between PageController.php and Page.php

Reply:
Agree here, perhaps we need to explicitly mention to still keep 1 class per file.

Reply:
I agree with Wilr. `Versioned` for example is a great example of putting too much in one file (yes it's all one class, but that class has too many concerns IMO). And _definitely_ we should explicitly say one class per file.

Maybe something like this
```suggestion
- Each class, interface, enum, and trait must be in its own file.
- Each class, interface, enum, and trait should adhere to the [single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle) where practical.
```

Reply:
Sorry I definitely meant one class per file, would fail PSR linting if you had more than one anyway I think.

Probably more a preference thing on my part against over abstractions, I'm not a fan of seeing a handful of 100 line files which would have easily been condensed into single file that would have been half the size and easier to comprehend.  I get that approach doesn't scale well though.

Reply:
Have removed line

Reply:
It's already covered by PSR-1 / PSR-12 - https://www.php-fig.org/psr/psr-1/#3-namespace-and-class-names